src/plugins/unified_field_list/public/hooks/use_grouped_fields.ts

```diff
@@ -68,33 +75,51 @@ export function useGroupedFields<T extends FieldListItem = DataViewField>({
     // if field existence information changed, reload the data view too
   }, [dataViewId, services.dataViews, setDataView, hasFieldDataHandler]);
 
+  // important when switching from a known dataViewId to no data view (like in text-based queries)
```

Comment:
When reading the code I noticed L66, it's the function `hasFieldDataByDefault` that always returns `true`?

Reply:
Yes, that's right. It's used as a fallback and does not cause useEffect reruns.